### PR TITLE
Week 9

### PR DESCRIPTION
**Pull request by payasbhrigu**
Animation

### CODE REVIEW

**Comment by payasbhrigu** [comment]
@anilrao316

File: Week-9/demo-react-app/src/Home.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            {/* <button onClick={() => removeNote(props.index)}>Delete</button> */}
+        </div>
+    );
+}
+
+function Header() {
+    return (
+        <header>
+            <h1 id="todoAppText">TodoApp</h1>
+        </header>
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/Home.js:27]: Create this as a seperate component and if any styles are there create a seperate file for it

File: Week-9/demo-react-app/src/Home.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect } from "react";
+import './Home.css'
+
+function Note(props){
+
+    function handleCheck(id){
+        var col = document.getElementById(id), check = document.getElementById(id+"_done")
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/Home.js:7]: This too create a seperate component like Note.js

File: Week-9/demo-react-app/.gitignore [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# dependencies
+/node_modules
+/.pnp
+.pnp.js
+
+# testing
+/coverage
+
+# production
+/build
```
Comment (anilrao316) [inline Week-9/demo-react-app/.gitignore:12]: Remove this file

File: Week-9/demo-react-app/src/Home.css [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    margin-top: 2px;
+    border: 2px solid black;
+    background-color: rgb(187, 201, 214)
+}
+
+input{
+    margin: 15px 5px 15px 8px;
+}
+
+#todoAppText{
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/Home.css:17]: Make use of class

File: Week-9/demo-react-app/src/Home.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        if(check.checked)
+            col.style.textDecoration = "line-through";
+        else
+            col.style.textDecoration = "initial";
+    }
+
+    return (
+        <div id={props.id} className="note">
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/Home.js:15]: Better to destructure props vakue at the top like const {id,noteText}=props

File: Week-9/demo-react-app/src/Home.js (hunk 2 above)
Comment (anilrao316) [reply Week-9/demo-react-app/src/Home.js:7]: U should  not use document.getelemnt.. in react very bad practice.. make use of useref hook to access any dom elemnet

File: Week-9/demo-react-app/src/TodoApp.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+export default function TodoApp(){
+    const [show, setShow] = useState(true)
+    const [count, setCount] = useState(delay)
+    
+    useEffect(() => {
+        let timer = setTimeout(() => {
+        setShow(false)
+        }, delay * 1000);
+
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/TodoApp.js:14]: What does thia file do?

File: Week-9/practice.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        } else {
+            left++
+            right++
+            return  [left,right]
+        }
+    }
+    return -1
+}
+
+console.log(twoSum([2,3,5,17,18],22) );
```
Comment (anilrao316) [inline Week-9/practice.js:26]: Remove all unrelated files

File: Week-9/demo-react-app/src/TodoApp.js (hunk 6 above)
Comment (anilrao316) [reply Week-9/demo-react-app/src/TodoApp.js:14]: Ok got it for loading on start

File: Week-9/demo-react-app/src/Home.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        <>
+            <Header/>
+            <div>
+                <input value={note['description']} type="text" placeholder="Enter New Note Here" name="note" onChange={(e) => setNote({description: e.target.value, done:false})}/>
+                <button onClick={addNote}>Add</button>
+            </div>
+            <div className="notes">
+                {notes.map((note,index) => {
+                    return <Note id={`note_${index}`} noteText={note['description']} index={index}/>
+                })}
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/Home.js:54]: Is index prop used plz check

File: Week-9/demo-react-app/src/App.test.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
+  render(<App />);
```
Comment (anilrao316) [inline Week-9/demo-react-app/src/App.test.js:5]: Try to add more test cases.. it will be good learning for u and try to add test cases for individual component

File: Week-9/demo-react-app/src/Home.js (hunk 8 above)
Comment (payasbhrigu) [reply Week-9/demo-react-app/src/Home.js:54]: Yes this is used in the delete function for note.

**Comment by payasbhrigu** [comment]
Updated Demo:
TodoApp

**Comment by anilrao316** [comment]
Looks good!
